[정동민] Assignment 1 - 초기 환경 세팅

sladhe01 opened the pull request:
## :: 최근 작업 주제 (하나 이상의 주제를 선택해주세요.)
- [x] 기능 추가
- [ ] 데이터베이스 작업
- [x] 리팩토링
- [ ] 버그 수정
- [x] 컨벤션 수정

<br />

## :: 구현 목표 - 브랜치(PR)에서 구현하고자 하는 하나의 목표 작성
- 모듈 설치 및 초기 세팅, 브랜치 생성, .gitignore 수정, typeorm dbconnection

<br />

## :: 구현 사항 설명 - 해당 브랜치(PR)에서 작업한 내용 작성
- express,  cors, dotenv, morgan, mysql2, typeorm 모듈 설치
- dongmin-initial-setting branch 생성
- .gitignore에 python,pycharm,visualstudiocode,visualstudio,vim,macos,zsh,linux,webstorm,windows,node 관련 파일 추가
- typeorm과 mysql database 연결
- **dongmin 폴더로 파일 경로 변경**
- **패키지 종류에 따라 순서대로 코드 변경**
- **server connection 기능 생성 및 dbconnection 순차적으로 실행되게 수정**
- **cors, morgan, express.json middleware 추가**
- **port번호를 변수로 지정**
<br />

## :: 테스트 결과 이미지
1. Server가 잘 동작하는지 확인할 수 있는 Terminal 캡쳐 이미지
2. Postman(Client Tool)을 이용한 API 테스트 결과 이미지
3. DB 작업 PR인 경우 Table 생성 및 수정 결과에 대해 확인할 수 있는 이미지
![image](https://github.com/wecode-bootcamp-korea/48-backend-westagram-4/assets/135579112/3afedb33-af41-4961-8324-521322d33f90)

## :: 기타 질문 및 특이 사항
- 제가 작성한 함수의 로직이 너무 복잡하다고 생각이 됩니다. 더 효율적인 방법이 없을까요? (예시)

Commit:
Feat: add modules, new branch

Install cors, dotenv, express, morgan, mysql2, typeorm

Commit:
docs: fix .gitignore

kyeongraekim commented on app.js:
```diff
@@ -0,0 +1,20 @@
+const express = require("express");
+const cors = require("cors");
+const dotenv = require("dotenv");
+const logger = require("morgan");
+
+const app = express();
+dotenv.config();
```
- 모듈 import 순서
    - import 순서 컨벤션에 맞게 수정해주세요! 사소한 부분이라 놓치기 쉽지만 처음 코드를 작성할 때부터 신경쓰고 작성하게 되면 추후에 관리하기도 용이하며 가독성도 좋아집니다.
    - 다음과 같은 순서로 import 해줍니다.
        - Bulit-in 패키지
        - 3rd-party 패키지
        - custom 패키지
        
        ```jsx
        //// Built-in package
        const http = require('http')
        
        //// 3rd-party package
        require('dotenv').config();
        const express = require('express');
        const cors    = require('cors');
        const morgan  = require('morgan');
        const { DataSource } = require('typeorm');
        
        //custom package
        const api = require('./../api')
        
        const app = express();
        ```

kyeongraekim commented on app.js:
```diff
@@ -0,0 +1,20 @@
+const app = express();
+dotenv.config();
+const { DataSource } = require("typeorm");
+const myDataSource = new DataSource({
```
- myDataSource 보다는 `appDataSource` , `mysqlDataSource` 혹은 간결하게 `database`로 변수명을 변경해주세요. “my”라는 용어는 예제 코드에서는 사용할 수 있으나 실제 코드에서는 피해주세요.
  - 추가로 `dataSource.initialize()` 함수가 실행되고 나면, then()과 catch()를 사용해서 에러를 핸들링해줄 수 있습니다. 이렇게 하면, initialize() 함수가 실행되어서 database와 실제로 connection을 생성할 때, 성공했는지 실패했는지 여부를 확인할 수 있습니다.

kyeongraekim commented on app.js:
```diff
@@ -0,0 +1,20 @@
+dotenv.config();
+const { DataSource } = require("typeorm");
+const myDataSource = new DataSource({
+  type: process.env.TYPEORM_CONNECTION,
+  host: process.env.TYPEORM_HOST,
+  port: process.env.TYPEORM_PORT,
+  username: process.env.TYPEORM_USERNAME,
+  password: process.env.TYPEORM_PASSWORD,
+  database: process.env.TYPEORM_DATABASE,
```
- 해당 TYPEORM_ 으로 시작하는 내용의 변수선언은, ORM 기능을 TypeORM에만 종속성이 생기는 모습 처럼 보여집니다. 
- 추후의 다른 라이브러리 사용성에 대한 확장성을 고려하여서, DB_CONNECTION 과 같은 외부 라이브러리 명칭이 아닌 데이터베이스의 성질을 띌 수 있는 변수로 이름을 바꾸어 주세요!~

kyeongraekim commented on app.js:
```diff
@@ -0,0 +1,20 @@
+  database: process.env.TYPEORM_DATABASE,
+});
+
+myDataSource.initialize().then(() => {
+  console.log("Data Source has been initialized!");
+});
```
- 현재 app.js 에 데이터베이스와의 연결만 확보가 되어있지, 해당 npm start 등의 본 서버의 동작을 요청하는 코드가 보이지 않네요!
- app.listen() 메소드를 이용하여, npm start 등의 서버 동작 명령어가 돌아갈 수 있게 세팅하여주세요!

kyeongraekim reviewed (changes requested):
현재 코드 수정 내역이 반영되어있지 않아 보입니다!~ 해당 내용 수정해서 재푸시해주세요!

Commit:
fix: Fix initial setting

Fixing app.js as changing package ordering, changing variable name myDataSource to appDataSource, changing variable names TypeORM_*** to DB_***, adding server creating function, adding PORT variable
Adding .env.sample file
Moving (app.js, package.json, package-lock.json) to ./
dongmin
Fixing .gitignore as adding schema.sql

Commit:
fix: Fix initial setting

Adding middleware(cors, morgan, express.json)

kyeongraekim reviewed (approved):
LGTM!~
